### ISSUE TITLE
Improve inline backend options

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#2040]
_From [dave.lin...@gmail.com](https://code.google.com/u/107110792395568123171/) on 2014-11-05T12:48:51Z_

Spyder Version:  2.3.1
Python Version:  3.4.2
Qt Version    :  4.8.6, PyQt4 (API v2) 4.10.4 on Windows
pyflakes >=0.6.0:  0.8.1 (OK)
pep8 >=0.6      :  1.5.7 (OK)
IPython >=0.13  :  2.3.0 (OK)
pygments >=1.6  :  1.6 (OK)
pandas >=0.13.1 :  0.15.0 (OK)
sphinx >=0.6.6  :  1.2.3 (OK)
rope >=0.9.2    :  0.9.4-1 (OK)
matplotlib >=1.0:  1.4.1 (OK)
sympy >=0.7.0   :  0.7.5 (OK)
pylint >=0.25   :  1.3.1 (OK) 

What steps will reproduce the problem?

 1. pylab.rcParams['figure.dpi'] = 150
pylab.rcParams['figure.figsize'] = 16, 12
plt.imshow(img)

pylab.rcParams['figure.dpi'] = 75
pylab.rcParams['figure.figsize'] = 16, 12
plt.imshow(img) 

What is the expected output? What do you see instead?

 I expect the first plot to be double the width of the second plot.

But both plots are exactly the same size. 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2040_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-18T14:41:34Z_

If you're using the IPython console, these options don't have effect. To change the size of images, you need to go to

Tools > Preferences > IPython Console > Graphics > Inline backend

and modify the dpi and figsize parameters there. You need to open a new console for these changes to take effect. Unfortunately, this is a limitation of the inline backend, i.e. when graphics are printed directly in the console.

Please report back if this fixes the problem for you.

**Labels:** MS-v2.3.3

**Comment by spyder-issues-migrator** [comment]
_From [dave.lin...@gmail.com](https://code.google.com/u/107110792395568123171/) on 2014-11-21T11:26:19Z_

Thanks; that has some effect, but there are still problems.

The default was 72 DPI and my physical monitor is 101.4 DPI (30" 2560x1600 LCD).  When I change the DPI to 101 (it doesn't allow fractions), then:

pylab.rcParams['figure.figsize'] = 8, 6

..gives me figures 6.2" x 4.6" (6."6 x 4.8" if you include the X and Y axis labels).

(That is supposed to give figures of 8x6 inches.)

Back-calculating, I tried setting 130 DPI (101 * 8/6.2) -- and it won't allow a setting higher than 112!  But at 112 it was indeed (112/101) bigger, so if I could have set it to 130, I should have gotten the right size figures.

To summarize:

a) Even with the DPI set to match the monitor in Tools>Preferences>IPython Console>Graphics>Inline backend, figures appear to be about 78% of the correct size.

b) It won't let you set the DPI higher than 112.

c) It doesn't allow fractional DPI (I'd like to set 101.4).

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-21T13:09:29Z_

As I said,

pylab.rcParams['figure.figsize']

doesn't have effect either. You need to change its value also using our Preferences system, changing the Width and Height values of the Inline backend.

I'll improve how we set dpi in our next release.

**Summary:** Improve inline backend options  \<span class="oldvalue"> (was: pylab.rcParams['figure.dpi'] has no effect) \</span>  
  **Status:** Accepted  
  **Labels:** Cat-IPythonConsole

**Comment by spyder-issues-migrator** [comment]
_From [dave.lin...@gmail.com](https://code.google.com/u/107110792395568123171/) on 2014-11-22T16:12:53Z_

Oh, but it does! This code works fine:

def small():
    pylab.rcParams['figure.figsize'] = 4, 3

def med():
    pylab.rcParams['figure.figsize'] = 8, 6
    
def big():
    pylab.rcParams['figure.figsize'] = 16, 12


If you type "small()", then plot something, you get a small plot. If you then type "big()" and plot it again, you'll get a big plot.

Just not the promised size in inches. But proportionally bigger and smaller, yes.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-02-01T08:26:59Z_

This issue was closed by revision 82788b3dd1be .

**Status:** Fixed
